### PR TITLE
Bump artifacts to latest release

### LINKED ISSUE #174: Bump dependencies to latest versions for a used major
I think we can be a bit more aggressive about bumping versions

currently thinking about 
- [ ] joda-time 2.9.3 => 2.9.9
- [ ] okhttp 2.4.0 => 2.7.5 
- [ ] okio 1.6.0 => 1.13.0
- [ ] okhttp3 3.6.0 => 3.7.0
- [ ] antlr4-runtime 4.5.3 => 4.7.0
- [ ] maven 3.3.9 => 3.5.0
- [ ] slf4j 1.7.16 => 1.7.25
- [ ] jsr305 1.3.9 => 3.0.1
- [ ] RoaringBitMap 0.5.11 => 0.6.43
- [ ] netty-all 4.0.43.Final => 4.1.9.Final
- [ ] netty 3.9.9.Final => 3.10.6.Final (last release)
- [ ] jetty 9.2.16.v20160414 => 9.4.4.v20170414
- [ ] jersey 2.22.2 => 2.25.1

### PR DESCRIPTION
**Pull request by robert3005**
Needs some offline testing. fixes #174

### CODE REVIEW

**Comment by robert3005** [comment]
@ash211 what do you think? Is there checkstyle test to verify things?

**Comment by robert3005** [comment]
`dev/lint-java` - will add to our regular builds

File: common/network-common/src/main/java/org/apache/spark/network/crypto/TransportCipher.java [hunk 1]
```diff
@@ -17,23 +17,26 @@
...
+import com.google.common.base.Preconditions;
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInboundHandlerAdapter;
+import io.netty.channel.ChannelOutboundHandlerAdapter;
+import io.netty.channel.ChannelPromise;
+import io.netty.channel.FileRegion;
+import io.netty.util.AbstractReferenceCounted;
```
Comment (ash211) [inline common/network-common/src/main/java/org/apache/spark/network/crypto/TransportCipher.java]: these import reorderings probably don't pass checkstyle

Comment (robert3005) [inline common/network-common/src/main/java/org/apache/spark/network/crypto/TransportCipher.java]: they do

**Comment by robert3005** [comment]
Had to slightly tweak grammar to fail in when parsing schema and there are some unconsumed characters.

I actually don't understand why test case of `a INT b long` is failing right now since it shouldn't. If you upgrade antlr in no longer fails so I added additional rule to ensure it's consistent

**Pull request merged**